src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/Internal/HostTests.cs

```diff
@@ -1373,7 +1434,9 @@ public ValueTask DisposeAsync()
 
         private class AsyncThrowingService : BackgroundService
         {
-            private readonly Task _executeDelayTask;
+            private readonly Task? _executeDelayTask;
+
+            public AsyncThrowingService() { }
```

Comment:
I think we should always be supplying an `executeDelayTask`, so we can control when the exception is thrown. We should remove this constructor and keep the field as non-nullable.

Reply:
This addition was very intentional. I wanted to control when the exception was thrown. Specifically, I wanted to test throwing immediately on the call to `StartAsync` as `StartAsync` was moved, and it's entirely possible for `StartAsync` to throw in the wild. Does that make sense?